File: frontend/src/types/product.ts
export interface Variation {
    id: string;
    name: string;
    external_id: string;
    show: boolean;
}

export interface VariationGroup {
    id: string;
    name: string;
    external_id: string;
    show: boolean;
    required: boolean;
    variations: Variation[];
}

export interface Product {
    id: string;
    external_id: string;
    name: string;
    slug: string;
    description: string | null;
    price: number;
    weight: number | null;
    sort: number;
    hidden: boolean;
    alcohol: boolean;
    sold: boolean;
    image: string;
    variation_groups: VariationGroup[];
}

File: backend/product-service/internal/models/product.go
package models

import (
	"github.com/google/uuid"
)

type Product struct {
	ID              uuid.UUID               `json:"id" db:"id"`
	ExternalID      string                  `json:"external_id" db:"external_id" validate:"max=100"`
	Name            string                  `json:"name" db:"name" validate:"required,min=2,255"`
	Slug            string                  `json:"slug" db:"slug" validate:"required,min=2,max=255"`
	Description     *string                 `json:"description" db:"description"`
	Price           float64                 `json:"price" db:"price" validate:"required,min=0"`
	Weight          *float64                `json:"weight" db:"weight" validate:"omitempty,min=0"`
	CategoryID      uuid.UUID               `json:"category_id" db:"category_id" validate:"required"`
	Sort            int                     `json:"sort" db:"sort" validate:"min=0"`
	Hidden          bool                    `json:"hidden" db:"hidden"`
	Alcohol         bool                    `json:"alcohol" db:"alcohol"`
	Sold            bool                    `json:"sold" db:"sold"`
	Image           string                  `json:"image" db:"image"`
	Stock           *float64                `json:"stock" db:"stock"`
	VariationGroups []ProductVariationGroup `json:"variation_groups,omitempty" db:"-"`
}
